source/agora/common/Task.d

```diff
@@ -73,14 +73,80 @@ public class TaskManager
             dg = This delegate will be called when the timer fires.
             periodic = Specifies if the timer fires repeatedly or only once.
 
+        Returns:
+            A `ITimer` instance with the ability to control the VibeTimer
     ***************************************************************************/
 
-    public void setTimer (Duration timeout, void delegate() dg,
+    public ITimer setTimer (Duration timeout, void delegate() dg,
         Periodic periodic = Periodic.No)
     {
         assert(dg !is null, "Cannot call this delegate if null");
-
         static import vibe.core.core;
-        vibe.core.core.setTimer(timeout, dg, periodic);
+        return new VibeTimer(vibe.core.core.setTimer(timeout, dg, periodic));
+    }
+}
+
+/*******************************************************************************
+
+    Define the ability to control the timer
+
+*******************************************************************************/
+
+public interface ITimer
+{
+    /***************************************************************************
+
+        Stop the timer executed by the setTimer
+
+    ***************************************************************************/
+
+    public void stop();
+}
+
+/*******************************************************************************
+
+    Vibe only timer
+
+*******************************************************************************/
+
+final class VibeTimer : ITimer
+{
+    static import Vibe = vibe.core.core;
+
+    private Vibe.Timer timer;
+
+    public this (Vibe.Timer timer)
+    {
+        this.timer = timer;
+    }
+
+    /// Ditto
+    public override void stop()
+    {
+        this.timer.stop();
+    }
+}
+
+/*******************************************************************************
+
+    LocalRest only timer (for unittests)
+
+*******************************************************************************/
+
+final class LocalRestTimer : ITimer
```

Comment:
This doesn't belong here, but in `Base.d` where it's used. And it should be `private` too.